assets/src/data/model/entity-factory/base-entity.js

```diff
@@ -127,13 +127,20 @@ class BaseEntity {
 	 * @return {BaseEntity} A new instance of BaseEntity
 	 */
 	get clone() {
-		return new BaseEntity(
-			this.modelName,
-			this.forClone,
-			{ $schema: {}, properties: this.schema },
-			this.fieldPrefixes,
-			true
-		);
+		return ( keepId = false ) => {
+			// @todo memoize this
+			const factory = createEntityFactory(
```

Comment:
I plan on memoizing this, forgot to add it for the pull.  Memoizing will ensure a factory is only ever created once per model here.